### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by Exper-t** [hashlips-lab/nft-erc721-collection#366] [Q&A]
After restarting 3 times on version 2.4.2 I have again got this error after entering "yarn verify"+contract address+" --network truffle"

This is the error I get:
**"Error in plugin @nomiclabs/hardhat-etherscan: Invalid API Key

For more info run Hardhat with --show-stack-traces
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command."**

However I don't see what is wrong with my API key as it is made from the Etherscan.io website just as Hashlips did in this video: https://www.youtube.com/watch?v=Zhmj4PiJ-GA

at the time 50:50

Could someone please help me solve this.

Many Thanks

**Comment by bolisteward** [comment]
I got the same issue, you could solve it ?

**Comment by liarco** [comment]
Hi @Exper-t and @bolisteward,
First of all that video from Daniel is great, but it's a bit outdated now (it's using v1), I recommend checking out the ones [listed here](https://github.com/hashlips-lab/nft-erc721-collection#youtube-tutorials) since they our latest ones.

API keys for block explorers must be created per chain (Etherscan API key works on Rinkeby and Goerli since they are all Ethereum chains, but it won't work for Polygon, Mumbai or BSC). Make sure you are using a correct API key for the chain you are working on.

Version [`2.4.3`](https://github.com/hashlips-lab/nft-erc721-collection/releases/tag/2.4.3) introduced a configuration bug that may have caused this same error for some users. That bug has been quickly fixed by [`2.4.4`](https://github.com/hashlips-lab/nft-erc721-collection/releases/tag/2.4.4) a couple of days later. If you don't want to start from scratch, you can simply replicate [these tiny changes](https://github.com/hashlips-lab/nft-erc721-collection/compare/2.4.3...2.4.4#diff-304d788adeada9a8d8914e56564ae8696b385fc877b62fba8262cea518e468b9).

Hope this helps.

**Reply by bolisteward** [reply]
thank you
